db/collection.go

```diff
@@ -25,6 +26,7 @@ type readOnlyCollection struct {
 	name      string
 	modelType reflect.Type
 	indexes   []*Index
+	indexMut  sync.RWMutex
```

Comment:
Since we're now copying the indexes whenever we create a snapshot, we need a mutex to protect them.